Integrations/integration-ThreatMiner.yml

```diff
@@ -0,0 +1,863 @@
+    threadMinerFoundResults = True
+
+  if threadMinerFoundResults == False:
+    md += 'No results found'
+
+  return md
+
+def get_ip_report_from_threat_miner(ipAddress):
+  whoisRawData = get_ip_whois_rawdata(ipAddress)
+  passiveDNSRawData = get_ip_passiveDNS_rawdata(ipAddress)
+  md5RawData = get_ip_MD5_rawdata(ipAddress)
+  uriRawData = get_ip_URI_rawdata(ipAddress)
+  sslRawData = get_ip_SSL_rawdata(ipAddress)
+  return  {
+  'raw_whois': whoisRawData,
+  'raw_passive_dns': passiveDNSRawData,
+  'raw_ip_md5': md5RawData,
+  'raw_ip_uris': uriRawData,
+  'raw_ip_ssl': sslRawData
+}
+
+def get_ip_context_format(threatMinerRawResults, maxReturnedArraySize, ipAddress):
+  ipContext = {
+  'ThreatMiner.IP.Address': ipAddress,
+  'ThreatMiner.IP.Whois': get_ip_whois(threatMinerRawResults, ipAddress),
+  'ThreatMiner.IP.PassiveDNS': get_ip_passiveDNS(threatMinerRawResults, maxReturnedArraySize),
+  'ThreatMiner.IP.MD5': get_ip_MD5(threatMinerRawResults, maxReturnedArraySize),
+  'ThreatMiner.IP.URI': get_ip_URI(threatMinerRawResults, maxReturnedArraySize),
+  'ThreatMiner.IP.SSL': get_ip_SSL(threatMinerRawResults, maxReturnedArraySize)
+}
+
+  return  createContext(ipContext, removeNull=True)
+
+def ip_command(maxReturnedArraySize):
+  ipAddress = demisto.args().get('ip')
+  if not is_ip_valid(ipAddress):
+    return_error('An invalid IP was specified')
+  threatMinerRawResults = get_ip_report_from_threat_miner(ipAddress)
+  threatMinerContext = get_ip_context_format(threatMinerRawResults, maxReturnedArraySize, ipAddress)
+  markdown = create_ip_command_markdown(ipAddress, threatMinerContext)
+  ipcontext = {
+  "IP.Address": threatMinerContext['ThreatMiner']['IP']['Address'],
+  "IP.Geo.Country": threatMinerContext['ThreatMiner']['IP']['Whois']['Country'],
+  "IP.ASN": threatMinerContext['ThreatMiner']['IP']['Whois']['ASN'],
+}
+  context = {
+  'ThreatMiner.IP(val.Address && val.Address == obj.Address)':  createContext(threatMinerContext['ThreatMiner']['IP'], removeNull=True),
+  'IP(val.Address && val.Address == obj.Address)': createContext(ipcontext, removeNull=True)
+}
+
+  demisto.results(
+  {'Type': entryTypes['note'],
+  'Contents': threatMinerRawResults,
+  'HumanReadable': markdown,
+  'EntryContext': context,
+  'ContentsFormat': formats['json']})
+
+def get_file_whois_rawdata(hashedFile):
+  threatMinerIpUrlPostfix = 'sample.php?q={}&rt={}'.format(hashedFile, 1)
+  response = http_request('GET',THREAT_MINER_URL + threatMinerIpUrlPostfix, DEFAULT_HEADERS)
+  threatminerResultsAsArray = response.get('results',[])
+
+  if len(threatminerResultsAsArray) == 0:
+    return threatminerResultsAsArray
+
+  return threatminerResultsAsArray[0]
+
+def get_file_whois_and_update_theartMinerFileContext(rawdataFile, context):
+  if len(rawdataFile['raw_whois']) == 0:
+    return rawdataFile['raw_whois']
+
+  context['ThreatMiner']['File']['MD5'] = rawdataFile['raw_whois']['md5']
```

Comment:
this should be like:
```
context = 
     'ThreatMiner.File(val.MD5 == obj.MD5)': {
             'MD5': rawdataFile['raw_whois']['md5'],
             'Sha1': rawdataFile['raw_whois']['sha1'],
            etc.....
      }
}
```

Reply:
I've tied the MD5 values but just outside the context function. Do I have to do it inside?